src/table.rs

```diff
@@ -61,40 +62,351 @@ impl Table {
+        for _ in 0..self.players.len() {
+            self.current_player_index += 1;
+            if self.current_player_index >= self.players.len() {
+                self.current_player_index = 0;
+            }
+            match self.players.get(self.current_player_index).unwrap().player_state {
+                PlayerState::Folded => {}
+                PlayerState::Active(_) => { break }
+            }
+        }
+    }
+
+    /// Deals cards for the flop, turn, and river
+    fn deal_table_cards(&mut self, deck_iterator: &mut Iter<Card>) {
+        self.flop = Option::from([
+            *deck_iterator.next().unwrap(),
             *deck_iterator.next().unwrap(),
             *deck_iterator.next().unwrap(),
         ]);
-
         self.turn = Option::from(*deck_iterator.next().unwrap());
         self.river = Option::from(*deck_iterator.next().unwrap());
+    }
 
+    /// Mark all players that died from the last round as dead now
+    fn check_for_player_death(&mut self) {
+        // Check if players died on the past round
         for player in &mut self.players {
-            let card1 = *deck_iterator.next().unwrap();
-            let card2 = *deck_iterator.next().unwrap();
-            player.deal([card1, card2])
+            if player.death_hand_number.is_none() && player.total_money < self.ante {
+                player.death_hand_number = Some(self.hand_number)
+            }
         }
     }
 
+    /// Deal cards to the alive players and collect the ante from them.
+    fn deal_player_cards_collect_ante(&mut self, deck_iterator: &mut Iter<Card>) {
+        // Deal every alive player cards now
+        for player in &mut self.players {
+            if player.is_alive() {
+                let card1 = *deck_iterator.next().unwrap();
+                let card2 = *deck_iterator.next().unwrap();
+                player.deal([card1, card2]);
+                self.pot += player.bet(self.ante);
+                // the ante doesn't count as a turn so clarify the bot hasn't had a turn
+                player.has_had_turn_this_round = false;
+            }
+        }
+    }
+
+    pub fn get_state_string_for_player(&self, id: i8) -> JsonValue {
+        let player_strings: Vec<_> = self.players.iter().map(|x| if x.get_id() == id { x.as_json() } else { x.as_json_no_secret_data() }).collect();
+        object! {
+            flop: self.get_flop_string_secret(),
+            turn: self.get_turn_string_secret(),
+            river: self.get_river_string_secret(),
+            dealer_button_index: self.dealer_button_index,
+            players: player_strings,
+            hand_number: self.hand_number,
+        }
+    }
+    fn is_betting_over(&self) -> bool {
+        let all_players_bet_or_folded = self.check_all_players_ready_for_next_round();
+        let all_players_equal_bet = self.check_all_active_players_same_bet();
+        all_players_bet_or_folded && all_players_equal_bet
+    }
+    fn get_max_bet(&self) -> i32 {
```

Comment:
This is fine once I read the function, but when I first saw calls to this, I was a bit confused thinking it meant the max amount players are allowed to bet. Not really sure what else I would call this though.

Reply:
Updated.